.github/workflows/rust.yml

```diff
@@ -76,12 +76,13 @@ jobs:
       - name: Checkout
         uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # @v3.1.0
 
-      # On macOS, we need a proper Clang version, not Apple's custom version without wasm32 support
+      # Install LLVM 15 on all platforms. On Linux and Windows, LLVM 16 is not compatible with
+      # the consensus/domain runtime build. On macOS, we need a proper Clang version, not Apple's
+      # custom version without wasm32 support.
```

Comment:
The old comment needs to remain as it was and second separate comment needs to be added that with TODO explaining that on non-macOS platforms this is a temporary workaround that we should remove once compilation is fixed.